scintillometry/utils/phase_utils.py

```diff
@@ -0,0 +1,127 @@
+# Licensed under the GPLv3 - see LICENSE
+
+import warnings
+
+import numpy as np
+import astropy.units as u
+import astropy.time as time
+
+try:
+    from .pint_utils import PintUtils
+except ImportError:
+    pass
+from .predictor import Polyco
+
+
+__all__ = ['PintPhase', 'PolycoPhase']
+
+
+class PintPhase(object):
+    """A utility class designated for computing phase using PINT.
+       Parameter
+       ---------
+       par_file : str
+           TEMPO/TEMPO2 style parameter fileself.
+       obs : str
+           Observatory name or observatory codeself.
+       obs_freq : float or `~astropy.units.Quantity`.
+           Observing frequency default units is 'MHz'
+       ephem : str
+           The solar system ephemeris version.
+       bipm_version : str
+           The BIPM clock correction version.
+
+       Note
+       ----
+       This method provides high precision phase calculation. The fractional
+       phase is between -0.5 to 0.5.
+    """
+    def __init__(self, par_file, obs, obs_freq, ephem='de436',
+                 bipm_version='BIPM2015'):
+        self.par_file = par_file
+        self.pu = PintUtils(self.par_file)
```

Comment:
I'm usually against local imports, but here I think it makes a lot of sense to just import `PintUtils` in here (and thus omit the import above)._